lib/configurator/index.js

```diff
@@ -0,0 +1,48 @@
+const EventEmitter = require('events');
+const hash = require('crypto').createHash;
+const storesFactory = require('./stores');
+
+class Configurator extends EventEmitter {
+  constructor(config) {
+    super();
+    const type = config.type;
+    this.logger = config.logger;
+    this.metrics = config.metrics;
+
+    this.store = config.store || new storesFactory(type, config);
+
+    this.pollInterval = config.interval || 300000; // 5 minutes
+    this.currentHash = hash('sha1').update(String(config.currentVal)).digest('base64');
+    this.startPolling();
+  }
+
+  checkForChanges() {
+    this.logger.info('Configurator: fetching config');
+    const start = Date.now();
+    this.store.fetch((err, data) => {
+      if (err) {
+        return this.logger.error(err);
```

Comment:
Add a log_type or a message or some way to identify where the message comes from